src/middleware/types.ts

```diff
@@ -413,17 +413,19 @@ export interface Option {
 }
 
 // TODO: there's a lot of repetition in the following type. factor out some common parts.
+// TODO: Should ack be different for Block actions?
 // tslint:disable:max-line-length
-type OptionsAckFn<Within extends 'interactive_message' | 'dialog_suggestion'> =
-  Within extends 'interactive_message' ?
+type OptionsAckFn<Container extends 'interactive_message' | 'dialog_suggestion' | ExternalSelectResponse> =
+  Container extends ('interactive_message' | ExternalSelectResponse) ?
     AckFn<{ options: { text: string; value: string; }[]; option_groups: { label: string; options: { text: string; value: string; }[]; }[]; }> :
   AckFn<{ options: { label: string; value: string; }[]; option_groups: { label: string; options: { label: string; value: string; }[]; }[]; }>;
 // tslint:enable:max-line-length
 
-export interface SlackOptionsMiddlewareArgs<Within extends 'interactive_message' | 'dialog_suggestion'> {
-  payload: ExternalOptionsRequest<Within>;
+export interface SlackOptionsMiddlewareArgs<Container extends 'interactive_message' |
+  'dialog_suggestion' | ExternalSelectResponse> {
+  payload: ExternalOptionsRequest<Container> | Actions<ExternalSelectResponse>;
```

Comment:
the union here is dangerous, because only one of these types is right for a given event, and there's no condition on which to determine which is right.

taking the `ack` property as an example, it uses `OptionsAckFn`, which is a conditional type to determine which of the two shapes are right for a given instantation of the generic parameter `Within`. we should try to create that relationship here too.